main.js

```diff
@@ -0,0 +1,138 @@
+const locationInput = document.getElementById("locationInput");
+
+async function fetchWeatherData(lat, lon) {
+  try {
+    const response = await fetch(
+      `https://api.openweathermap.org/data/3.0/onecall?lat=${lat}&lon=${lon}&exclude=minutely,hourly&appid=${apiKey}&units=imperial`
+    );
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error fetching weather data:", error);
+  }
+}
+
+async function getApproximateLocation() {
+  const response = await fetch("http://ip-api.com/json");
+  const data = await response.json();
+  return {
+    lat: data.lat,
+    lon: data.lon,
+    name: `${data.city}, ${data.regionName}, ${data.country}`,
+    zip: data.zip,
+    timezone: data.timezone,
+  };
+}
+
+function displayCurrentWeather(data, cityName) {
+  const { temp, weather } = data.current;
+
+  document.getElementById("cityName").innerText = cityName;
+  document.getElementById("temperature").innerText = `${temp.toFixed(1)}°F`;
+  document.getElementById("weatherDescription").innerText =
+    weather[0].description;
+  document.getElementById(
+    "weatherIcon"
+  ).src = `http://openweathermap.org/img/wn/${weather[0].icon}.png`;
+}
+
+function displayForecast(daily) {
+  forecastDiv.innerHTML = "";
+  daily.slice(1, 6).forEach((day) => {
+    const date = new Date(day.dt * 1000);
+    const options = { weekday: "long" };
+    const dayName = date.toLocaleDateString("en-US", options);
+    const { temp, weather } = day;
+    forecastDiv.innerHTML += `
+          <div class="col forecast-card">
+              <div class="card">
+                  <div class="card-body text-center">
+                      <h3>${dayName}</h3>
+                      <p>${temp.day.toFixed(1)}°F</p>
+                      <p>${weather[0].description}</p>
+                      <img src="http://openweathermap.org/img/wn/${
+                        weather[0].icon
+                      }.png" alt="${weather[0].description}">
+                  </div>
+              </div>
+          </div>`;
+  });
+}
+
+function isValidZipCode(zip) {
+  if (zip.length === 5) {
+    return zip.split("").every((char) => char >= "0" && char <= "9");
+  } else if (zip.length === 10) {
+    return (
+      zip[5] === "-" &&
+      zip
+        .slice(0, 5)
+        .split("")
+        .every((char) => char >= "0" && char <= "9") &&
+      zip
+        .slice(6)
+        .split("")
+        .every((char) => char >= "0" && char <= "9")
+    );
+  }
+  return false;
+}
+
+async function handleWeatherSearch() {
+  const location = locationInput.value.trim();
+  if (!location) return;
+
+  let geoUrl;
+  if (isValidZipCode(location)) {
+    geoUrl = `https://api.openweathermap.org/geo/1.0/zip?zip=${location}&appid=${apiKey}`;
+  } else {
+    geoUrl = `https://api.openweathermap.org/geo/1.0/direct?q=${location}&limit=1&appid=${apiKey}`;
+  }
```

Comment:
you can use a ternary operator here
```suggestion
  const geoUrl = isValidZipCode(location) ? `https://api.openweathermap.org/geo/1.0/zip?zip=${location}&appid=${apiKey}` :  `https://api.openweathermap.org/geo/1.0/direct?q=${location}&limit=1&appid=${apiKey}`;
```